modelopt/torch/_compress/tools/bypassed_training/child_init.py

```diff
@@ -0,0 +1,1649 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# mypy: ignore-errors
+
+"""TODO Add description. Analyze this code, why is it so long and complex? Can it be simplified?"""
+
+import concurrent.futures
+import dataclasses
+import json
+import os
+import re
+import time
+from copy import deepcopy
+from enum import Enum
+from functools import partial
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+try:
+    from typeguard import check_type
+except ImportError:
+    print("Could not import typeguard. Type checking is skipped.")
+
+    def check_type(x, cls):
+        return x
```

Comment:
Can we just add `typeguard` dependency to compress dependencies in setup.py and remove the import guard?